yt_dlp/extractor/youtube.py

```diff
@@ -4322,7 +4414,7 @@ def process_manifest_format(f, proto, client_name, itag, po_token):
 
             # Clients that require PO Token return videoplayback URLs that may return 403
             # hls does not currently require PO Token
-            if (not po_token and self._get_default_ytcfg(client_name).get('REQUIRE_PO_TOKEN')) and proto != 'hls':
+            if (not po_token and _PoTokenContext.GVS in self._get_default_ytcfg(client_name)['PO_TOKEN_REQUIRED_CONTEXTS']) and proto != 'hls':
```

Comment:
just curious why we need the parentheses. is it a ruff thing? it's just `if x and y and z` right?

Reply:
dunno, fixed